### PR TITLE
feat: Issue number 197 add v2 api favorite mixes

### PR DESCRIPTION
**Pull request by jozefKruszynski**
I ended up having to create a MixV2 class as the structure is different enough to warrant it, however, this structure is only used for this one v2 api endpoint, and tidal themselves in their web app use the v1 api for every other call related to mixes.

It's a rather odd situation as you need to use the v2 endpoint to return a list of something that you then actually only ever need the name and ID of really.

I'll push a draft PR to make what I'm talking about a bit more tangible.

### CODE REVIEW

**Comment by tehkillerbee** [comment]
@jozefKruszynski  Thanks for the PR. Can this be moved from draft or is more work needed on it?

**Comment by jozefKruszynski** [comment]
I left it as a draft, as I was hoping for some feedback on the code.
I'll remove the draft as, I guess you'll be checking the code anyway.

**Comment by jozefKruszynski** [comment]
I'll fix the conflict later this evening, as I'm at work at the moment

**Comment by tehkillerbee** [comment]
@jozefKruszynski  Looks like the typing additions I just merged resulted in new conflicts. Sorry!

**Comment by jozefKruszynski** [comment]
Having a hard time getting my changes to work as they did previously. I think I'm losing my mind

**Comment by jozefKruszynski** [comment]
Realised I had completely overcomplicated the whole thing. I have pushed a much simplified version

File: tidalapi/user.py [hunk 1]
```diff
@@ -380,3 +383,18 @@ def videos(self) -> List["Video"]:
...
+    def mixes(self, limit: Optional[int] = 50, offset: int = 0) -> List["MixV2"]:
+        """Get the users favorite tracks.
+
+        :return: A :class:`list` of :class:`~tidalapi.media.Track` objects containing all of the favorite tracks.
+        """
+        params = {"limit": limit, "offset": offset}
+        return cast(
+            List["MixV2"],
+            self.requests.map_request(
+                url=urljoin("https://api.tidal.com/v2", f"{self.v2_base_url}/mixes"),
```
Comment (tehkillerbee) [inline tidalapi/user.py:396]: Doing a simple test results in a 404 HTTP Error
```
mixes = self._active_session.user.favorites.mixes()
...
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.tidal.com/favorites/mixes?sessionId=XXX&countryCode=DK&limit=50&offset=0
```
A quick glance reveals that v2 is omitted in the url so V2 url is invalid.

urljoin removes the `v2` part. It looks like we need to add a forward slash. Adding it fixes the issue.

```suggestion
                url=urljoin("https://api.tidal.com/v2/", f"{self.v2_base_url}/mixes"),
```

**Review by tehkillerbee — CHANGES_REQUESTED** [review]

File: tidalapi/user.py (hunk 1 above)
Comment (jozefKruszynski) [reply tidalapi/user.py:396]: Confused, I'm certain that test passed on my machine when I was running it.

**Comment by tehkillerbee** [comment]
@jozefKruszynski  

A sidenote, after looking more into the tidal2 plugin for KODI, I noticed that there are a few endpoints that use the V2 API

```
    def get_followers(self, user_id, offset=0, limit=500):
        return self._map_request_v2('profiles/%s/followers' % user_id, params={'deviceType': 'BROWSER'}, ret='userprofiles')

    def get_following_users(self, user_id, offset=0, limit=500):
        return self._map_request_v2('profiles/%s/following' % user_id, params={'includeOnly': 'USER', 'deviceType': 'BROWSER'}, ret='userprofiles')

    def get_following_artists(self, user_id, offset=0, limit=500):
        return self._map_request_v2('profiles/%s/following' % user_id, params={'includeOnly': 'ARTIST', 'deviceType': 'BROWSER'}, ret='artists')

    def get_public_playlists(self, user_id, offset=0, limit=50):
        return self._map_request_v2('user-playlists/%s/public' % user_id, params={'offset': offset, 'limit': limit}, ret='playlists')
```

File: tidalapi/user.py (hunk 1 above)
Comment (tehkillerbee) [reply tidalapi/user.py]: Yeah its quite odd. Maybe a different version of urllib?

**Comment by tehkillerbee** [comment]
Works as expected and tests pass. Merged!

**Comment by jozefKruszynski** [comment]
> @jozefKruszynski  
> 
> 
> 
> A sidenote, after looking more into the tidal2 plugin for KODI, I noticed that there are a few endpoints that use the V2 API
> 
> 
> 
> ```
> 
>     def get_followers(self, user_id, offset=0, limit=500):
> 
>         return self._map_request_v2('profiles/%s/followers' % user_id, params={'deviceType': 'BROWSER'}, ret='userprofiles')
> 
> 
> 
>     def get_following_users(self, user_id, offset=0, limit=500):
> 
>         return self._map_request_v2('profiles/%s/following' % user_id, params={'includeOnly': 'USER', 'deviceType': 'BROWSER'}, ret='userprofiles')
> 
> 
> 
>     def get_following_artists(self, user_id, offset=0, limit=500):
> 
>         return self._map_request_v2('profiles/%s/following' % user_id, params={'includeOnly': 'ARTIST', 'deviceType': 'BROWSER'}, ret='artists')
> 
> 
> 
>     def get_public_playlists(self, user_id, offset=0, limit=50):
> 
>         return self._map_request_v2('user-playlists/%s/public' % user_id, params={'offset': offset, 'limit': limit}, ret='playlists')
> 
> ```

I guess a robust V2 solution is not what I have produced at all. We can look into a proper implementation in the near future I hope.
Work is a bitch at the moment

**Pull request merged**